bpf/bpf_lxc.c

```diff
@@ -1303,16 +1303,17 @@ int tail_ipv6_to_ipv4(struct __ctx_buff *ctx)
 {
 	int ret = ipv6_to_ipv4(ctx, 14, LXC_IPV4);
 	if (IS_ERR(ret))
-		return  send_drop_notify(ctx, SECLABEL, 0, 0, ret, CTX_ACT_DROP,
-				METRIC_EGRESS);
+		goto drop_err;
 
 	cilium_dbg_capture(ctx, DBG_CAPTURE_AFTER_V64, ctx->ingress_ifindex);
 
 	ctx_store_meta(ctx, CB_NAT46_STATE, NAT64);
 
 	invoke_tailcall_if(__and(is_defined(ENABLE_IPV4), is_defined(ENABLE_IPV6)),
 			   CILIUM_CALL_IPV4_FROM_LXC, tail_handle_ipv4);
-	return ret;
+drop_err:
+	return send_drop_notify(ctx, SECLABEL, 0, 0, ret, CTX_ACT_DROP,
+				METRIC_EGRESS);
```

Comment:
I don't think this is correct.

Reply:
`invoke_tailcall_if ()` results into a normal/inlined call if the v4/v6 condition is not satisfied, so this means we always send a drop notification for these cases, no?

Reply:
@borkmann If the conditions are not met, it returns as far as I can see, so we won't hit the `send_drop_notify()`:
https://github.com/cilium/cilium/blob/1dc3e396c1e048c94ad65d6f6bdb0463bf3281fb/bpf/lib/tailcall.h#L90-L91

Reply:
Good point, you're right! Merging then. 👍